javascripts/discourse/components/sidebar-new-topic-button.gjs

```diff
@@ -74,12 +78,30 @@ export default class SidebarNewTopicButton extends Component {
     this.tag = this.router.currentRoute.attributes?.tag || null;
   }
 
+  @action
+  watchForComposer() {
+    // this covers opening drafts from the hamburger menu
+    this.appEvents.on("composer:will-open", this, this.closeHamburger);
+  }
+
+  @action
+  stopWatchingForComposer() {
+    this.appEvents.off("composer:will-open", this, this.closeHamburger);
+  }
+
+  @action
+  closeHamburger() {
+    this.header.hamburgerVisible = false;
+  }
+
   <template>
     {{#if this.shouldRender}}
       <div
         class="sidebar-new-topic-button__wrapper"
         {{didInsert this.getCategoryandTag}}
         {{didUpdate this.getCategoryandTag this.router.currentRoute}}
+        {{didInsert this.watchForComposer}}
+        {{willDestroy this.stopWatchingForComposer}}
```

Comment:
The `willDestroy` event fires when the sidebar is removed from the DOM, yes? I don't quite follow why we to mark `hamburgerVisible = false` in that case.

Reply:
Ah, that makes sense. Thanks!